### PR TITLE
AddressSanitizer: heap-buffer-overflow in PwlApply32

### PR DESCRIPTION
**Pull request by dmitriikhurtin**
### Details:
 - Fixed a reading error outside the allocated memory range

### Tickets:
 - CVS-73168

### CODE REVIEW

**Review by mryzhov — APPROVED** [review]

**Review by elilobanova — APPROVED** [review]

File: src/plugins/intel_gna/gna_graph_compiler.cpp [hunk 1]
```diff
@@ -882,6 +882,18 @@ void GNAGraphCompiler::PoolingPrimitive(InferenceEngine::CNNLayerPtr layer) {
     uint32_t num_padding = ALIGN(hw_in, 8) - hw_in;
     size_t num_data_bytes_in = c_dim_in * (hw_in + num_padding) * inputs->getPrecision().size();
 
+    if (dnn->new_num_conv_columns) {
```
Comment (elilobanova) [inline src/plugins/intel_gna/gna_graph_compiler.cpp:885]: this code seems to be the same as for PWL primitive, can it be shared with a separate function?

Comment (mryzhov) [inline src/plugins/intel_gna/gna_graph_compiler.cpp:885]: I understand that everybody knows what new_num_conv_columns  mean, but could you please add the short comment, personally for me :)

Comment (dmitriikhurtin) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:885]: In the line 379 you can find few comments about new_num_conv_columns and an algorithm that calculates it. In general, @elilobanova, please, correct me if I'm wrong, this variable is used to store an output of convolution taking into account padding.

Comment (dmitriikhurtin) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:885]: The same only this code:
if (dnn->new_num_conv_columns % num_columns == 0) {
num_rows = dnn->new_num_conv_columns / num_columns;
} else {
num_columns = dnn->new_num_conv_columns;
}
I think it's unnecessary to create a function for a four lines code.

Comment (mryzhov) [reply src/plugins/intel_gna/gna_graph_compiler.cpp:885]: Line 379 is a part of another function and it is not clear here why we are processing it (dnn->new_num_conv_columns) here. Could you please comment such logic next time

**Pull request merged**